runtime/jcl/common/jdk_internal_vm_Continuation.cpp

```diff
@@ -65,4 +67,28 @@ Java_jdk_internal_vm_Continuation_unpin(JNIEnv *env, jclass unused)
 	}
 }
 
+void JNICALL
+Java_jdk_internal_vm_Continuation_setFinishedImpl(JNIEnv *env, jobject recv)
+{
+	J9VMThread *vmThread = (J9VMThread *) env;
+//	enterVMFromJNI(vmThread);
+	j9object_t object = J9_JNI_UNWRAP_REFERENCE(recv);
+	ContinuationState *continuationStatePtr = VM_VMHelpers::getContinuationStateAddress(vmThread, object);
+	VM_VMHelpers::setContinuationFinished(continuationStatePtr);
+//	exitVMToJNI(vmThread);
+}
+
+jboolean JNICALL
+Java_jdk_internal_vm_Continuation_isFinishedImpl(JNIEnv *env, jobject recv)
+{
+	jboolean result = JNI_FALSE;
+	J9VMThread *vmThread = (J9VMThread *) env;
+	j9object_t object = J9_JNI_UNWRAP_REFERENCE(recv);
+	ContinuationState *continuationStatePtr = VM_VMHelpers::getContinuationStateAddress(vmThread, object);
+	if (VM_VMHelpers::isFinished(*continuationStatePtr)) {
+		result = JNI_TRUE;
+	}
+	return result;
+}
+
```

Comment:
I guess, these do not really have to be implemented in native - it's just by doing so, all the API that assumes the knowledge of how the state is organized is concentrated at one spot. We could really get/set this bit in java code too, right?

More generally, I'd slightly prefer not to remove start/finish bools as a part of these changes to minimize the scope, since I'd consider this for 0.37 propagation, but I'd let VM reviewers to decide.